frontend/lib/with-signed-in.tsx

```diff
@@ -0,0 +1,24 @@
+import React from "react"
+import { NextPageContext as NextContext } from "next"
+import { isSignedIn } from "/lib/authentication"
+import redirect from "/lib/redirect"
+
+export default function withSignedIn(Component: any) {
+  return class WithSignedIn extends React.Component {
+    static displayName = `withSignedIn(${Component.displayName ?? "Component"})`
+
+    static async getInitialProps(context: NextContext) {
+      if (!isSignedIn(context)) {
+        redirect(context, "/sign-in")
```

Comment:
If we're redirecting to a different page, we probably should skip fetching initial props and return an empty object here.

Reply:
It does now.